src/server/pps/server/api_server.go

```diff
@@ -3605,3 +3567,49 @@ func ensurePipelineProject(p *pps.Pipeline) {
 		p.Project = &pfs.Project{Name: pfs.DefaultProjectName}
 	}
 }
+
+func jqFilterFunc(ctx context.Context, filter string) (func(proto.Message) (bool, error), error) {
```

Comment:
Closing over a context is a lot like stashing it in a struct — something to be avoided unless using an intermediate context-ignorant library. It is better to pass the context through the call chain, so that each function can attenuate the context properly.

Since `jqFilterFunc` doesn’t really do anything requiring a context and is unexported, it doesn’t need to be passed in as an argument; it may always be added later. It’s not _wrong_ to include it, just not necessary.

Concretely, this should be `func jqFilterFunc(string) (func(context.Context, proto.Message) (bool, error), error)` and the call sites should look like `jqFilter(ctx, msg)`.

I believe that the name here is not so great either: this isn’t really a “jq filter func”; it _compiles_ a jq  filter to a filter func. I would probably call this `newFilter` or `newFilterFromJQ`. Since this is all unexported, `newFilter` or `newMessageFilter` is probably fine for now, but it’s up to you.

It’s probably not necessary to declare a type for the return value, although it too wouldn’t be wrong.